### ISSUE TITLE
Add or remove curves using Checkbox

### ISSUE THREAD

**Issue by hayg25** [taurus-org/taurus#1153]
Hi everyone, 
is there a way to add or remove curves on TaurusTrend by using for example a TaurusValueCheckBox. 

thanks

**Comment by cpascual** [comment]
Hi, first, just to be able to be more precise, can you specify if you are talking about the qwt5 (old implementation) or the pyqtgraph based TaurusTrend? 

Apart from that, you mention a `TaurusValueCheckBox`, but I do not see how that precise widget can be used here (a `TaurusValueCheckBox` is used to write the value of a boolean taurus attribute). AFAICT, what you probably want is a pure Qt `QCheckBox`. For this you should create your own subclass that stores the curve model name and connects its `stateChanged`(or `toggled`) signal to a method that adds/removes that curve from the trend (or hides/shows it, whatever you prefer).

If you are using the qwt5 trend , a similar thing is already implemented by clicking on the legend ([clicking a curve in the legend toggles it between Y1 axis, Y2 axis and hidden](http://taurus-scada.org/users/ui/plot.html#working-with-two-y-scales)).
In the new taurus_pyqtgraph application this is not yet implemented, but it has been already asked, so it could be done as well

**Comment by hayg25** [comment]
Hi @cpascual, 
thanks for your answer. Yes I realized after my post that clicking on the legend works fine and that's exactly what I need. So nothing to develop there :) 
one more question : I am actually using qwt5 TaurusTrend. Do you see advantages on using pyqtgraph based taurustrend ?
same question for the other plotting  (images, etc)
Many thanks !

**Comment by cpascual** [comment]
I am glad to hear that what is already implemented works for your use case.

Regarding the qwt5 vs pyqtgraph implementations:

- qwt5 only works with python2 + pyqt4  . Python 2, will become unsupported in taurus > 4.7 and PyQt4 is being removed from most modern distributions.
- the qwt5 plots/trends are no longer actively developed in taurus (they are quite stable, but no new features are added and only critical bugs have chances of being fixed)
- taurus_pyqtgraph lacks some features that exist in the old qwt5 implementations (such as the "click-on-legend"), but since taurus_pyqtgraph v0.4.0 I consider the new implementations ready for replacing the old ones in the vast majority of cases. In some aspects they have even more features. Our plans are to make tauruspyqtgraph the default before the end of this year (see #452 )

In conclusion, my advice is to test replacing your old plots with the newer implementation, and unless you find some blocker issue with them, use them.
